From: David Henningsson <david.henningsson@canonical.com>
To: pulseaudio-discuss@lists.freedesktop.org
Cc: fsateler@debian.org,
	David Henningsson <david.henningsson@canonical.com>
Subject: [PATCH] build-sys: Don't build srbchannel-test on targets without srbchannel
Date: Tue,  2 Dec 2014 09:31:11 +0100
Message-Id: <1417509071-9050-1-git-send-email-david.henningsson@canonical.com>
X-Mailer: git-send-email 1.9.1

Reported-by: Felipe Sateler <fsateler@debian.org>
Signed-off-by: David Henningsson <david.henningsson@canonical.com>
---

So, I prefer having skipping the srbchannel test instead of having a succeding one
in case there is no srbchannel functionality. Felipe, does this patch your issue?

 configure.ac    | 1 +
 src/Makefile.am | 6 +++++-
 2 files changed, 6 insertions(+), 1 deletion(-)

--- a/configure.ac
+++ b/configure.ac
@@ -449,6 +449,7 @@ AC_CHECK_HEADERS_ONCE([execinfo.h])
 AC_CHECK_HEADERS_ONCE([langinfo.h])
 AC_CHECK_HEADERS_ONCE([regex.h pcreposix.h])
 
+AM_CONDITIONAL(HAVE_SYS_EVENTFD_H, test "x$ac_cv_header_sys_eventfd_h" = "xyes")
 
 #### Typdefs, structures, etc. ####
 
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -288,11 +288,15 @@ TESTS_daemon = \
 
 if !OS_IS_WIN32
 TESTS_default += \
-		srbchannel-test \
 		sigbus-test \
 		usergroup-test
 endif
 
+if HAVE_SYS_EVENTFD_H
+TESTS_default += \
+		srbchannel-test
+endif
+
 if !OS_IS_DARWIN
 TESTS_default += \
 		once-test
